webpack/assets/javascripts/react_app/routes/RegistrationCommands/RegistrationCommandsPage/components/fields/RepositoryModalInput.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+import PropTypes from 'prop-types';
+
+import { FormGroup, TextInput, Grid } from '@patternfly/react-core';
+
+const RepositoryModalInput = ({
+  reference,
+  updateRepository,
+  isLoading,
+  initRepo,
+  initGpgKey,
+}) => {
+  const onRepoUpdate = repo => {
+    updateRepository(reference, repo, 'repository');
+  };
+
+  const onGpgKeyUpdate = gpgKey => {
+    updateRepository(reference, gpgKey, 'gpgKeyUrl');
+  };
+
+  return (
+    <Grid hasGutter md={6}>
+      <FormGroup fieldId="reg_repo">
+        <TextInput
+          ouiaId="reg_host_repo"
+          id="reg_host_repo"
+          key="reg_host_repo"
+          value={initRepo}
+          type="text"
+          onChange={onRepoUpdate}
+          isDisabled={isLoading}
+        />
+      </FormGroup>
+      <FormGroup fieldId="reg_gpg_key_url">
+        <TextInput
+          ouiaId="reg_host_gpg_key"
+          id="reg_host_gpg_key"
+          key="reg_host_gpg_key"
+          value={initGpgKey}
+          type="text"
```

Comment:
Is gpg key url a required field or I can also add  just repositories and leave the gpg url field blank? 
Does it need to have fixed format and should it have check for that?

Reply:
About the check, it is the same as for the repo URL. We could add it later, there is not check in the current implantation, either.